cobbler/cobbler_collections/collection.py

```diff
@@ -101,7 +106,9 @@ def deserialize_running(self, deserialize_running: bool):
         """
         self._deserialize_running = deserialize_running
 
-    def factory_produce(self, api, seed_data):
+    def factory_produce(
+        self, api: "CobblerAPI", seed_data: Dict[str, Any]
```

Comment:
You could fix the unused variable errors by using something like:

```
del api # unused
del seed_data # unused
```

Reply:
Since this is an abstract method I am going to leave that for now. :)